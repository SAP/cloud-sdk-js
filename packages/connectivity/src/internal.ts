export * from './http-agent';
export * from './scp-cf';
