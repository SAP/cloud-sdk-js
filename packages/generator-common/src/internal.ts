export * from './sdk-metadata';
export * from './compiler';
export * from './readme';
export * from './reserved-words';
export * from './util';
export * from './file-writer';
export * from './package-description';
export * from './ts-config';
export * from './options-parser';
export * from './cli-parser';
export * from './options';
export * from './options-per-service';
