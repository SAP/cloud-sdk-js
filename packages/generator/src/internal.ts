export * from './operations';
export * from './aggregator-package';
export * from './batch';
export * from './complex-type';
export * from './edmx-parser';
export * from './edmx-to-vdm';
export * from './entity';
export * from './enum-type';
export * from './request-builder';
export * from './sdk-metadata';
export * from './service';
export * from './swagger-parser';
export * from './file-generator';
export * from './generator-options';
export * from './generator-utils';
export * from './generator';
export * from './global-name-formatter';
export * from './imports';
export * from './input-path-provider';
export * from './internal-prefix';
export * from './name-formatting-strategies';
export * from './options-parser';
export * from './reserved-words';
export * from './service-generator';
export * from './service-mapping';
export * from './service-name-formatter';
export * from './typedoc';
export * from './vdm-types';

// Had to be added for public API check
export * from './generator-without-ts-morph';
