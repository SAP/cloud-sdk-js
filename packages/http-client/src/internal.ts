export * from './csrf-token-middleware';
export * from './http-client';
export * from './http-client-types';
export * from './http-request-config';
