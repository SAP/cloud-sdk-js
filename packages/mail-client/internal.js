'use strict';
function __export(m) {
  for (const p in m) {
    if (!exports.hasOwnProperty(p)) {
      exports[p] = m[p];
    }
  }
}
Object.defineProperty(exports, '__esModule', { value: true });
__export(require('./dist/internal'));
// # sourceMappingURL=internal.js.map
