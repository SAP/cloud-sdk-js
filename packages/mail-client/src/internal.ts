export * from './mail-client';
export * from './mail-client-types';
export * from './socket-proxy';
