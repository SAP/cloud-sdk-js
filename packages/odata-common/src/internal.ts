export * from './de-serializers';
export * from './filter';
export * from './order';
export * from './request';
export * from './request-builder';
export * from './selectable';
export * from './uri-conversion';
export * from './entity-builder';
export * from './response-transformer';
export * from './properties-util';
export * from './batch-response';
export * from './response-data-accessor';
export * from './edm-types';
export * from './entity-base';
export * from './entity-api';
export * from './time';
export * from './expandable';
export * from './entity-serializer';
export * from './entity-deserializer';
export * from './header-builder';
