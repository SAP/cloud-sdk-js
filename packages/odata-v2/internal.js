'use strict';
function __export(m) {
  for (const p in m) {
    if (!exports.hasOwnProperty(p)) {
      exports[p] = m[p];
    }
  }
}
Object.defineProperty(exports, '__esModule', { value: true });
/**
 * @packageDocumentation
 * @experimental The internal module is related to sdk-metadata types which are used only internally.
 */
__export(require('./dist/internal'));
// # sourceMappingURL=internal.js.map
