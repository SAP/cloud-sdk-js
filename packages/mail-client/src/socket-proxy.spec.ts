import { customAuthRequestHandler } from "../internal";

describe('socket proxy', () => {
  describe('customAuthRequestHandler', () => {
    it('returns the correct buffer without location ID', async () => {
      const jwt = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c';
      const locationId = undefined;

      const authRequest = await customAuthRequestHandler(jwt, locationId)

      expect(authRequest).toMatchSnapshot();
    });

    it('return the correct buffer with location ID', async () => {
      const jwt = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c';
      const locationId = 'cloud-sdk-js';

      const authRequest = await customAuthRequestHandler(jwt, locationId)

      expect(authRequest).toMatchSnapshot();
    });
  })
})