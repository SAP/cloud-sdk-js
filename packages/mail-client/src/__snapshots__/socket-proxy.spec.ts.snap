// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`socket proxy customAuthRequestHandler return the correct buffer with location ID 1`] = `
{
  "data": [
    1,
    0,
    0,
    0,
    155,
    101,
    121,
    74,
    104,
    98,
    71,
    99,
    105,
    79,
    105,
    74,
    73,
    85,
    122,
    73,
    49,
    78,
    105,
    73,
    115,
    73,
    110,
    82,
    53,
    99,
    67,
    73,
    54,
    73,
    107,
    112,
    88,
    86,
    67,
    74,
    57,
    46,
    101,
    121,
    74,
    122,
    100,
    87,
    73,
    105,
    79,
    105,
    73,
    120,
    77,
    106,
    77,
    48,
    78,
    84,
    89,
    51,
    79,
    68,
    107,
    119,
    73,
    105,
    119,
    105,
    98,
    109,
    70,
    116,
    90,
    83,
    73,
    54,
    73,
    107,
    112,
    118,
    97,
    71,
    52,
    103,
    82,
    71,
    57,
    108,
    73,
    105,
    119,
    105,
    97,
    87,
    70,
    48,
    73,
    106,
    111,
    120,
    78,
    84,
    69,
    50,
    77,
    106,
    77,
    53,
    77,
    68,
    73,
    121,
    102,
    81,
    46,
    83,
    102,
    108,
    75,
    120,
    119,
    82,
    74,
    83,
    77,
    101,
    75,
    75,
    70,
    50,
    81,
    84,
    52,
    102,
    119,
    112,
    77,
    101,
    74,
    102,
    51,
    54,
    80,
    79,
    107,
    54,
    121,
    74,
    86,
    95,
    97,
    100,
    81,
    115,
    115,
    119,
    53,
    99,
    0,
    99,
    108,
    111,
    117,
    100,
    45,
    115,
    100,
    107,
    45,
    106,
    115,
  ],
  "type": "Buffer",
}
`;

exports[`socket proxy customAuthRequestHandler returns the correct buffer without location ID 1`] = `
{
  "data": [
    1,
    0,
    0,
    0,
    155,
    101,
    121,
    74,
    104,
    98,
    71,
    99,
    105,
    79,
    105,
    74,
    73,
    85,
    122,
    73,
    49,
    78,
    105,
    73,
    115,
    73,
    110,
    82,
    53,
    99,
    67,
    73,
    54,
    73,
    107,
    112,
    88,
    86,
    67,
    74,
    57,
    46,
    101,
    121,
    74,
    122,
    100,
    87,
    73,
    105,
    79,
    105,
    73,
    120,
    77,
    106,
    77,
    48,
    78,
    84,
    89,
    51,
    79,
    68,
    107,
    119,
    73,
    105,
    119,
    105,
    98,
    109,
    70,
    116,
    90,
    83,
    73,
    54,
    73,
    107,
    112,
    118,
    97,
    71,
    52,
    103,
    82,
    71,
    57,
    108,
    73,
    105,
    119,
    105,
    97,
    87,
    70,
    48,
    73,
    106,
    111,
    120,
    78,
    84,
    69,
    50,
    77,
    106,
    77,
    53,
    77,
    68,
    73,
    121,
    102,
    81,
    46,
    83,
    102,
    108,
    75,
    120,
    119,
    82,
    74,
    83,
    77,
    101,
    75,
    75,
    70,
    50,
    81,
    84,
    52,
    102,
    119,
    112,
    77,
    101,
    74,
    102,
    51,
    54,
    80,
    79,
    107,
    54,
    121,
    74,
    86,
    95,
    97,
    100,
    81,
    115,
    115,
    119,
    53,
    99,
    0,
  ],
  "type": "Buffer",
}
`;
